bin/emqx

```diff
@@ -124,6 +124,20 @@ generate_config() {
         APPCONF=`relx_nodetool mergeconf $RUNNER_ETC_DIR/emqx.conf $RUNNER_ETC_DIR/plugins $RUNNER_DATA_DIR/configs`
         replace_env_in_conf
         CONFIG_ARGS=`$CUTTLEFISH_COMMAND_PREFIX -c $APPCONF generate`
+
+        ## Merge cuttlefish generated *.args into the vm.args
+        CUTTLE_GEN_ARG_FILE=`echo "$CONFIG_ARGS" | sed -n 's/^.*\(vm_args[[:space:]]\)//p' | awk '{print $1}'`
+        TMP_ARG_FILE="$RUNNER_DATA_DIR/configs/vm.args.tmp"
+        cp "$RUNNER_ETC_DIR/vm.args" "$TMP_ARG_FILE"
+        sed '/^#/d' $CUTTLE_GEN_ARG_FILE | sed '/^$/d' | while IFS='' read -r ARG_LINE || [ -n "$ARG_LINE" ]; do
+            ARG_KEY=`echo "$ARG_LINE" | awk '{print $1}'`
+            ARG_VAL=`echo "$ARG_LINE" | awk '{print $2}'`
+            ## if found, replace it, otherwise append a new line
+            grep -q "^$ARG_KEY" "$TMP_ARG_FILE" && \
```

Comment:
Do not mess around with `&&` and `||` since you are not really writing a one-liner.
use `if` `then` `else` ftw.
```
if grep -q "^$ARG_KEY" "$TMP_ARG_FILE"; then
  sed "s/^$ARG_KEY.*/$ARG_LINE/" -i "$TMP_ARG_FILE"
else
  echo "$ARG_LINE" >> "$TMP_ARG_FILE"
fi
```